src/util/apiHealth.ts

```diff
@@ -0,0 +1,67 @@
+import { fetchWithProxy } from '../fetch';
+import { CloudConfig } from '../globalConfig/cloud';
+
+export interface HealthResponse {
+  status: string;
+  message: string;
+}
+
+/**
+ * Checks the health of the remote API.
+ * @param url - The URL to check.
+ * @returns A promise that resolves to the health check response.
+ */
+export async function checkRemoteHealth(url: string): Promise<HealthResponse> {
+  try {
+    const cloudConfig = new CloudConfig();
+    const response = await fetchWithProxy(url, {
```

Comment:
sorry now that I see this, use `fetchWithTimeout` which calls `fetchWithProxy`.  Then you don't need your own timeout logic.